lib/msal-browser/test/cache/BrowserCacheManager.spec.ts

```diff
@@ -325,7 +326,532 @@ describe("BrowserCacheManager tests", () => {
+                    const testVal = "testId";
+                    browserSessionStorage.setRedirectRequestContext(testVal);
+                    expect(browserSessionStorage.getRedirectRequestContext()).toEqual(testVal);
+                });
+
+            });
+        });
+    });
+
+    
+    describe("Interface functions with overridden temporaryCacheLocation", () => {
+        let browserSessionStorage: BrowserCacheManager;
+        let authority: Authority;
+        let browserLocalStorage: BrowserCacheManager;
+        let cacheVal: string;
+        let msalCacheKey: string;
+        let msalCacheKey2: string;
+        beforeEach(() => {
+            browserSessionStorage = new BrowserCacheManager(TEST_CONFIG.MSAL_CLIENT_ID, cacheConfig, browserCrypto, logger);
+            authority = new Authority(TEST_CONFIG.validAuthority, StubbedNetworkModule, browserSessionStorage, {
+                protocolMode: ProtocolMode.AAD,
+                authorityMetadata: "",
+                cloudDiscoveryMetadata: "",
+                knownAuthorities: []
+            }, logger);
+            sinon.stub(Authority.prototype, "getPreferredCache").returns("login.microsoftonline.com");
+            cacheConfig.cacheLocation = BrowserCacheLocation.LocalStorage;
+            cacheConfig.temporaryCacheLocation = BrowserCacheLocation.LocalStorage;
+            browserLocalStorage = new BrowserCacheManager(TEST_CONFIG.MSAL_CLIENT_ID, cacheConfig, browserCrypto, logger);
+            cacheVal = "cacheVal";
+            msalCacheKey = browserSessionStorage.generateCacheKey("cacheKey");
+            msalCacheKey2 = browserSessionStorage.generateCacheKey("cacheKey2");
+        });
+
+        afterEach(async () => {
+            await browserSessionStorage.clear();
+            await browserLocalStorage.clear();
+        });
+
+        it("setTemporaryCache", () => {
+            browserSessionStorage.setTemporaryCache("cacheKey", cacheVal, true);
+            browserLocalStorage.setTemporaryCache("cacheKey2", cacheVal, true);
+            expect(window.sessionStorage.getItem(msalCacheKey)).toBe(cacheVal);
+            expect(window.localStorage.getItem(msalCacheKey2)).toBe(cacheVal);
+        });
+
+        it(
+            "getTemporaryCache falls back to local storage if not found in session/memory storage",
+            () => {
+                const testTempItemKey = "test-temp-item-key";
+                const testTempItemValue = "test-temp-item-value";
+                window.localStorage.setItem(testTempItemKey, testTempItemValue);
+                cacheConfig.cacheLocation = BrowserCacheLocation.LocalStorage;
+                browserLocalStorage = new BrowserCacheManager(TEST_CONFIG.MSAL_CLIENT_ID, cacheConfig, browserCrypto, logger);
+                expect(browserLocalStorage.getTemporaryCache(testTempItemKey)).toBe(testTempItemValue);
+            }
+        )
+
+        it("setItem", () => {
```

Comment:
This test doesn't appear relevant to this PR (probably already exists)

Reply:
Agreed